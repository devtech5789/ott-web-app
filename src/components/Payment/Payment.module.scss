@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';

.infoBox {
  display: flex;
  justify-content: space-between;
  margin-bottom: variables.$base-spacing;
  padding: calc(#{variables.$base-spacing} / 2) variables.$base-spacing;

  font-size: 14px;
  line-height: 18px;
  text-shadow: 0 2px 4px rgba(0, 0, 0, 0.14), 0 3px 4px rgba(0, 0, 0, 0.12), 0 1px 5px rgba(0, 0, 0, 0.2);
  background: theme.$panel-bg;
  border-radius: 4px;

  > strong {
    line-height: 16px;
    letter-spacing: 0.25px;
  }
}

.transactionItem {
  display: flex;
  flex-direction: column;
  gap: 0.25em;

  > span {
    color: variables.$gray-light;
  }
}

.pendingSwitch {
  display: inline-block;
  margin-top: variables.$base-spacing;
}

.transactionDetails {
  display: flex;
  justify-content: center;
  align-items: center;
  > div {
    margin-left: variables.$base-spacing;
  }
}

.transationPrice {
  font-size: 1.25em;
}

.price {
  font-size: 14px;
  line-height: 18px;
  > strong {
    font-weight: var(--body-font-weight-bold);
    font-size: 24px;
    line-height: 26px;
  }
}

.cardDetails {
  display: flex;
  margin-top: calc(#{variables.$base-spacing} * 2);
}

.paypal {
  display: flex;
  align-items: center;
  margin-bottom: variables.$base-spacing;
  font-weight: var(--body-font-weight-bold);
  font-size: 24px;
  font-style: italic;

  > svg {
    margin-right: 4px;
  }
}

.upgradeSubscription {
  margin-right: 8px;

  @include responsive.mobile-only() {
    margin: 0 0 8px;
  }
}

.changePlanContainer {
  display: flex;
  flex-direction: column;
  gap: 24px;
}

.changePlanButtons {
  display: flex;
  flex-direction: row;
  width: 100%;
  gap: 12px;
}

.changePlanCancelButton {
  margin-left: auto;
}
