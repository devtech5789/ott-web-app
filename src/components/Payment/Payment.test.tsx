import React from 'react';

import Payment from './Payment';

import customer from '#test/fixtures/customer.json';
import transactions from '#test/fixtures/transactions.json';
import paymentDetail from '#test/fixtures/paymentDetail.json';
import subscription from '#test/fixtures/subscription.json';
import type { Customer } from '#types/account';
import type { PaymentDetail, Subscription, Transaction } from '#types/subscription';
import { renderWithRouter } from '#test/testUtils';

describe('<Payment>', () => {
  test('renders and matches snapshot', () => {
    const { container } = renderWithRouter(
      <Payment
        accessModel="AVOD"
        customer={customer as Customer}
        transactions={transactions as Transaction[]}
        activeSubscription={subscription as Subscription}
        activePaymentDetail={paymentDetail as PaymentDetail}
        pendingOffer={null}
        canUpdatePaymentMethod={false}
        showAllTransactions={false}
        isLoading={false}
        offerSwitchesAvailable={false}
        onShowReceiptClick={vi.fn()}
        onUpgradeSubscriptionClick={vi.fn()}
        onShowAllTransactionsClick={vi.fn()}
        changeSubscriptionPlan={{
          isLoading: false,
          isSuccess: false,
          isError: false,
          reset: vi.fn(),
        }}
        onChangePlanClick={vi.fn()}
        selectedOfferId={null}
        setSelectedOfferId={vi.fn()}
        isUpgradeOffer={undefined}
        setIsUpgradeOffer={vi.fn()}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
