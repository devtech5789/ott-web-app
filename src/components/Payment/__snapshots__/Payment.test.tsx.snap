// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Payment> > renders and matches snapshot 1`] = `
<div>
  <div>
    <div>
      <h3>
        user:payment.payment_method
      </h3>
    </div>
    <div>
      <div
        class="_textField_e16c1b"
      >
        <label
          class="_label_e16c1b"
          for="text-field_1235"
        >
          user:payment.card_number
        </label>
        <p>
          •••• •••• •••• 0002
        </p>
      </div>
      <div
        class="_cardDetails_c57ff5"
      >
        <div
          class="_textField_e16c1b"
        >
          <label
            class="_label_e16c1b"
            for="text-field_1235"
          >
            user:payment.expiry_date
          </label>
          <p>
            03/2030
          </p>
        </div>
        <div
          class="_textField_e16c1b"
        >
          <label
            class="_label_e16c1b"
            for="text-field_1235"
          >
            user:payment.security_code
          </label>
          <p>
            ******
          </p>
        </div>
      </div>
      <button
        class="_button_f8f296 _default_f8f296 _outlined_f8f296"
      >
        <span>
          account:payment.edit_card
        </span>
      </button>
    </div>
  </div>
  <div>
    <div>
      <h3>
        user:payment.billing_history
      </h3>
    </div>
    <div
      class="_infoBox_c57ff5"
    >
      <p
        class="_transactionItem_c57ff5"
      >
        <strong>
          May 5, 2021
        </strong>
        <span>
          Monthly subscription (recurring)
        </span>
        <span>
          T712014024
        </span>
      </p>
      <div
        class="_transactionDetails_c57ff5"
      >
        <div
          class="_transationPrice_c57ff5"
        >
          user:payment.price_paid_with
        </div>
      </div>
    </div>
    <div
      class="_infoBox_c57ff5"
    >
      <p
        class="_transactionItem_c57ff5"
      >
        <strong>
          May 5, 2021
        </strong>
        <span>
          Monthly subscription (recurring)
        </span>
        <span>
          T177974068
        </span>
      </p>
      <div
        class="_transactionDetails_c57ff5"
      >
        <div
          class="_transationPrice_c57ff5"
        >
          user:payment.price_paid_with
        </div>
      </div>
    </div>
    <div
      class="_infoBox_c57ff5"
    >
      <p
        class="_transactionItem_c57ff5"
      >
        <strong>
          May 5, 2021
        </strong>
        <span>
          Monthly subscription (recurring)
        </span>
        <span>
          T996601696
        </span>
      </p>
      <div
        class="_transactionDetails_c57ff5"
      >
        <div
          class="_transationPrice_c57ff5"
        >
          user:payment.price_paid_with
        </div>
      </div>
    </div>
  </div>
</div>
`;
